--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -370,6 +370,10 @@
   ${SDL2_SOURCE_DIR}/src/video/*.c
   ${SDL2_SOURCE_DIR}/src/video/yuv2rgb/*.c)
 
+if(IOS)
+   set(SOURCE_FILES ${SOURCE_FILES} ${SDL2_SOURCE_DIR}/src/render/metal/SDL_render_metal.m)
+endif()
+
 
 if(ASSERTIONS STREQUAL "auto")
   # Do nada - use optimization settings to determine the assertion level
@@ -637,6 +641,10 @@
 #  endif()
 endif()
 
+if(NOT IOS)
+  list(APPEND EXTRA_LIBS SDL2main)
+endif()
+
 # TODO: Can't deactivate on FreeBSD? w/o LIBC, SDL_stdinc.h can't define
 # anything.
 if(LIBC)
@@ -757,7 +765,10 @@
 endif()
 if(SDL_SENSOR)
   file(GLOB SENSOR_SOURCES ${SDL2_SOURCE_DIR}/src/sensor/*.c)
-  set(SOURCE_FILES ${SOURCE_FILES} ${SENSOR_SOURCES})
+  if(IOS)
+    file(GLOB EXTRA_SENSOR_SOURCES ${SDL2_SOURCE_DIR}/src/sensor/coremotion/*.m)
+  endif()
+  set(SOURCE_FILES ${SOURCE_FILES} ${SENSOR_SOURCES} ${EXTRA_SENSOR_SOURCES})
 endif()
 if(SDL_POWER)
   file(GLOB POWER_SOURCES ${SDL2_SOURCE_DIR}/src/power/*.c)
@@ -1482,7 +1493,7 @@
     find_library(IOKIT IOKit)
     list(APPEND EXTRA_LIBS ${IOKIT})
   endif()
-  if(SDL_FRAMEWORK_FF)
+  if(SDL_FRAMEWORK_FF AND NOT IOS)
     find_library(FORCEFEEDBACK ForceFeedback)
     list(APPEND EXTRA_LIBS ${FORCEFEEDBACK})
   endif()
@@ -1621,8 +1632,18 @@
 #   endif()
 # endif()
 
-configure_file("${SDL2_SOURCE_DIR}/include/SDL_config.h.cmake"
-  "${SDL2_BINARY_DIR}/include/SDL_config.h")
+# Since check_function_exists isn't working properly on iOS,
+# the auto-generated SDL_config.h falsely determines several
+# functions exist when they actually don't. As a workaround,
+# use the pre-generated header instead.
+if(IOS)
+  configure_file("${SDL2_SOURCE_DIR}/include/SDL_config_iphoneos.h"
+    "${SDL2_BINARY_DIR}/include/SDL_config.h"
+  )
+else()
+  configure_file("${SDL2_SOURCE_DIR}/include/SDL_config.h.cmake"
+    "${SDL2_BINARY_DIR}/include/SDL_config.h")
+endif()
 
 # Prepare the flags and remove duplicates
 if(EXTRA_LDFLAGS)
@@ -1738,6 +1759,20 @@
   set_target_properties(SDL2main PROPERTIES DEBUG_POSTFIX ${SDL_CMAKE_DEBUG_POSTFIX})
 endif()
 
+if(IOS)
+  set(IOS_FRAMEWORKS
+    "-framework AVFoundation"
+    "-framework CoreGraphics"
+    "-framework CoreMotion"
+    "-framework Foundation"
+    "-framework GameController"
+    "-framework Metal"
+    "-framework OpenGLES"
+    "-framework QuartzCore"
+    "-framework UIKit"
+  )
+endif()
+
 if(SDL_SHARED)
   add_library(SDL2 SHARED ${SOURCE_FILES} ${VERSION_SOURCES})
   if(APPLE)
@@ -1767,6 +1802,9 @@
   if (NOT ANDROID)
     set_target_properties(SDL2 PROPERTIES DEBUG_POSTFIX ${SDL_CMAKE_DEBUG_POSTFIX})
   endif()
+  if(IOS)
+    target_link_libraries(SDL2 ${IOS_FRAMEWORKS})
+  endif()
 endif()
 
 if(SDL_STATIC)
@@ -1792,6 +1830,9 @@
   target_include_directories(SDL2-static PUBLIC "$<BUILD_INTERFACE:${SDL2_SOURCE_DIR}/include>" $<INSTALL_INTERFACE:include/SDL2>)
   if (NOT ANDROID)
     set_target_properties(SDL2-static PROPERTIES DEBUG_POSTFIX ${SDL_CMAKE_DEBUG_POSTFIX})
+  endif()
+  if(IOS)
+    target_link_libraries(SDL2-static ${IOS_FRAMEWORKS})
   endif()
 endif()
 
